File: CS/DevExpress.OutlookInspiredApp.Wpf/Common/View/Behaviors/ExpandSelectedRowBehavior.cs
using System;
using System.Linq;
using System.Windows;
using DevExpress.Mvvm.UI.Interactivity;
using DevExpress.Xpf.Grid;

namespace DevExpress.DevAV.Common.View {
    public class ExpandSelectedRowBehavior : Behavior<GridControl> {
        protected override void OnAttached() {
            base.OnAttached();
            AssociatedObject.Loaded += OnAssociatedObjectLoaded;
            if(AssociatedObject.IsLoaded)
                ExpandSelectedRow();
        }
        void OnAssociatedObjectLoaded(object sender, RoutedEventArgs e) {
            ExpandSelectedRow();
        }
        protected override void OnDetaching() {
            base.OnDetaching();
            AssociatedObject.Loaded -= OnAssociatedObjectLoaded;
        }
        void ExpandSelectedRow() {
            int[] selectedRowHandles = AssociatedObject.GetSelectedRowHandles();
            if(selectedRowHandles != null && selectedRowHandles.Length > 0) {
                int handle = selectedRowHandles.First();
                AssociatedObject.ExpandMasterRow(handle);
            }
        }
    }
}

File: VB/DevExpress.OutlookInspiredApp.Wpf/Common/View/Behaviors/ExpandSelectedRowBehavior.vb
Imports System
Imports System.Linq
Imports System.Windows
Imports DevExpress.Mvvm.UI.Interactivity
Imports DevExpress.Xpf.Grid

Namespace DevExpress.DevAV.Common.View
    Public Class ExpandSelectedRowBehavior
        Inherits Behavior(Of GridControl)

        Protected Overrides Sub OnAttached()
            MyBase.OnAttached()
            AddHandler AssociatedObject.Loaded, AddressOf OnAssociatedObjectLoaded
            If AssociatedObject.IsLoaded Then
                ExpandSelectedRow()
            End If
        End Sub
        Private Sub OnAssociatedObjectLoaded(ByVal sender As Object, ByVal e As RoutedEventArgs)
            ExpandSelectedRow()
        End Sub
        Protected Overrides Sub OnDetaching()
            MyBase.OnDetaching()
            RemoveHandler AssociatedObject.Loaded, AddressOf OnAssociatedObjectLoaded
        End Sub
        Private Sub ExpandSelectedRow()
            Dim selectedRowHandles() As Integer = AssociatedObject.GetSelectedRowHandles()
            If selectedRowHandles IsNot Nothing AndAlso selectedRowHandles.Length > 0 Then
                Dim handle As Integer = selectedRowHandles.First()
                AssociatedObject.ExpandMasterRow(handle)
            End If
        End Sub
    End Class
End Namespace
